tests/integration/memgraph.cpp

```diff
@@ -0,0 +1,140 @@
+}
+
+std::string GetStringValue(const mg_value *value) {
+  const mg_string *s = mg_value_string(value);
+  return GetStringValue(s);
+}
+
+class MemgraphConnection : public ::testing::Test {
+ protected:
+  virtual void SetUp() override {
+    params = mg_session_params_make();
+    std::string memgraph_host =
+        GetEnvOrDefault<std::string>("MEMGRAPH_HOST", "127.0.0.1");
+    int memgraph_port = GetEnvOrDefault<int>("MEMGRAPH_PORT", 7687);
+    bool memgraph_ssl = GetEnvOrDefault<bool>("MEMGRAPH_SSLMODE", true);
+
+    mg_session_params_set_host(params, memgraph_host.c_str());
+    mg_session_params_set_port(params, memgraph_port);
+    mg_session_params_set_sslmode(
+        params, memgraph_ssl ? MG_SSLMODE_REQUIRE : MG_SSLMODE_DISABLE);
+  }
+
+  virtual void TearDown() override {
+    mg_result *result;
+    mg_session_run(session, "MATCH (n) DETACH DELETE n", NULL, NULL);
+    mg_session_pull(session, &result);
+    mg_session_params_destroy(params);
+    if (session) {
+      mg_session_destroy(session);
+    }
+  }
+
+  mg_session_params *params;
+  mg_session *session;
+};
+
+TEST_F(MemgraphConnection, CanConnectToMemgraph) {
+  int status = mg_connect(params, &session);
+
+  ASSERT_EQ(status, 0);
+}
+
+TEST_F(MemgraphConnection, InsertAndRetriveFromMemegraph) {
+  mg_result *result;
+  int status = 0, rows = 0;
+  const char *create_query =
+      "\
+        CREATE (n: TestLabel{id: 1, name: 'test1', is_deleted: true}) \
+        CREATE (n)-[:TestRel {attr: 'attr1'}]->(: TestLabel{id: 12, name: 'test2', is_deleted: false}) \
+    ";
+  const char *get_query =
+      "\
+        MATCH (n)-[r]->(m) \
+        RETURN n, r, m \
+    ";
+
+  mg_connect(params, &session);
+  mg_session_run(session, create_query, NULL, NULL);
+  mg_session_pull(session, &result);
+  mg_session_run(session, get_query, NULL, NULL);
+
+  while ((status = mg_session_pull(session, &result)) == 1) {
+    const mg_list *mg_columns = mg_result_columns(result);
+    const mg_list *mg_row = mg_result_row(result);
+
+    auto lsa = mg_value_get_type(mg_list_at(mg_columns, 0));
+    auto lsaa = GetStringValue(mg_list_at(mg_columns, 0));
+
+    EXPECT_EQ(GetStringValue(mg_list_at(mg_columns, 0)), "n");
+    EXPECT_EQ(GetStringValue(mg_list_at(mg_columns, 1)), "r");
+    EXPECT_EQ(GetStringValue(mg_list_at(mg_columns, 2)), "m");
+
+    const mg_node *node_n = mg_value_node(mg_row->elements[0]);
+    const mg_node *node_m = mg_value_node(mg_row->elements[2]);
+    const mg_relationship *relationship_r =
+        mg_value_relationship(mg_row->elements[1]);
+
+    EXPECT_EQ(GetStringValue(mg_node_label_at(node_n, 0)), "TestLabel");
+    EXPECT_EQ(GetStringValue(mg_relationship_type(relationship_r)), "TestRel");
+    EXPECT_EQ(GetStringValue(mg_node_label_at(node_m, 0)), "TestLabel");
+
+    // Assert properties of Node n
+    const mg_map *properties_n = mg_node_properties(node_n);
+    EXPECT_EQ(mg_value_integer(mg_map_at(properties_n, "id")), 1);
```

Comment:
As above, first verify the size of the map before accessing it.
Also, verify that the element indeed is an integer before casting it to an integer.